scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/language/json/JsonTemplateParser.scala

```diff
@@ -0,0 +1,181 @@
+  ): ValidatedNel[ExpressionParseError, String] = expression match {
+    case expression: CompositeStringExpression =>
+      expression.getExpressions.toList
+        .map(e => joinLiteralsAndReplaceExpressionWithDefaults(spelExpressionToDefaultJsonValue, e))
+        .sequence
+        .map(_.mkString)
+    case expression: LiteralExpression => validNel(expression.getValue)
+    case expression: standard.SpelExpression =>
+      spelExpressionToDefaultJsonValue(expression.getExpressionString, Typed[SpelExpressionRepr])
+    case _ => invalidNel(ExpressionCompilationError("Unknown expression type"))
+  }
+
+  private def spelExpressionToDefaultJsonValue(ctx: ValidationContext)(
+      original: String,
+      expectedType: typing.TypingResult
+  ): ValidatedNel[ExpressionParseError, String] = spelParser
+    .parse(original, ctx, expectedType)
+    .map(_.typingInfo.typingResult)
+    .map(typingResultToDefaultJsonValue)
+
+  private def typingResultToDefaultJsonValue(typingResult: TypingResult): String = typingResult match {
+    case TypedClass(k, _) =>
+      k match {
+        case className if TypeValueDeterminer.isIntegerNumber(className)       => "0"
+        case className if TypeValueDeterminer.isFloatingPointNumber(className) => "0.5"
+        case className if TypeValueDeterminer.isBoolean(className)             => "true"
+        case className if TypeValueDeterminer.isString(className)              => "unquoted string"
+        // For now, complex types are treated as String
+        case _ => "unquoted string"
+      }
+    // For now, complex types are treated as String
+    case _ => "unquoted string"
+  }
+
+  private def extractJsonStringFromCompiledExpression(
+      compiledExpression: CompiledExpression
+  ): ValidatedNel[ExpressionParseError, String] =
+    compiledExpression match {
+      case expression: SpelExpression =>
+        joinLiteralsAndReplaceExpressionWithDefaults(
+          spelExpressionToDefaultJsonValue,
+          expression.parsedSpringExpression,
+        )
+      case _ => invalidNel(ExpressionCompilationError("Invalid compiled expression type"))
+    }
+
+  private def spelExpressionToDefaultJsonValue(
+      original: String,
+      expectedType: typing.TypingResult
+  ): ValidatedNel[ExpressionParseError, String] = spelParser
+    .parseWithoutContextValidation(original, expectedType)
+    .map(_ => "0")
+
+}
+
+object JsonTemplateParser {
+  private val stringTypingResult = Typed.typedClass[String]
+
+  case class CompiledJsonTemplateExpression(
+      languageId: Language,
+      originalJsonString: String,
+      templateCompiledExpression: CompiledExpression,
+      expectedType: typing.TypingResult
+  ) extends CompiledExpression {
+
+    override def language: Language = languageId
+
+    override def original: String = originalJsonString
+
+    override def evaluate[T](ctx: Context, globals: Map[String, Any]): T =
+      if (expectedType == stringTypingResult) {
```

Comment:
IMO it is not a good behaviour for `String`. This:
```
"foo"
```
is a valid json. Also this:
```
123
```
is a valid json. This is too:
```
null
```
Let's say that someone know that in the passed json template, will be a string. They wanted to get just the `foo` (without quotes). Thanks to current "magic" it will be quoted `"foo"`. If we remove this if, everything should work as expected. For now, I don't see requirement to add such a magic so following YAGNI/KISS rules, let's remove it.